rs/canister_sandbox/src/sandbox_manager.rs

```diff
@@ -171,11 +171,14 @@ impl Execution {
                             // to delay the drop until after the execution
                             // completed message is sent back to the main
                             // process.
-                            Ok(mut instance) => instance
-                                .store_data_mut()
-                                .system_api_mut()
-                                .expect("System api not present in the wasmtime instance")
-                                .take_system_state_changes(),
+                            Ok(mut instance) => {
+                                let system_api_impl = instance
+                                    .store_data_mut()
+                                    .system_api_mut()
+                                    .expect("System api not present in the wasmtime instance");
+                                system_api_impl.update_on_low_wasm_memory_hook_status();
+                                system_api_impl.take_system_state_changes()
+                            }
```

Comment:
One alternative will be to call `update_on_low_wasm_memory_hook_status` inside `take_system_state_changes` so we can keep changes local to SystemApiImpl, and every time someone take system_state_changes they will have updated hook status. Any preferences?